doc/nrf/libraries/networking/nrf_cloud_log.rst

```diff
@@ -0,0 +1,163 @@
+
+Multiple dictionary-based log messages are sent together as a binary message to the `d2c/bin device message topic <nRF Cloud MQTT Topics_>`_.
+The nRF Cloud website will not display the contents of the dictionary-based log messages in real time.
+Instead, you must download a binary file containing the logs over a certain range of time, then decode the logs using a Python script and the dictionary built when the application was built.
+
+Requirements
+************
+
+The device must be connected to nRF Cloud before calling the :c:func:`nrf_cloud_log_send` function.
+The :c:func:`nrf_cloud_rest_log_send` function initiates the connection as needed.
+
+Configuration
+*************
+
+Configure one or both of the following options to enable direct log messages or the logging backend:
+
+* :kconfig:option:`CONFIG_NRF_CLOUD_LOG_DIRECT`
+* :kconfig:option:`CONFIG_NRF_CLOUD_LOG_BACKEND`
+
+If only the first is enabled:
+
+* Calls to the direct log message functions :c:func:`nrf_cloud_log_send()` and :c:func:`nrf_cloud_rest_log_send()` send messages direct to nRF Cloud immediately.
+* The cloud logging backend is not available and consequently no Zephyr log messages are transmitted to the cloud.
+
+If only the second is enabled:
+
+* Only the logging backend will be present and all enabled Zephyr log messages of the proper level will be sent to the cloud.
+* Direct log message functions are not available.
+
+If both options are enabled, calls to the direct log message functions are passed to the logging backend instead.
+
+Configure one of the following options to select the data transport method:
+
+* :kconfig:option:`CONFIG_NRF_CLOUD_MQTT` or :kconfig:option:`CONFIG_NRF_CLOUD_REST`
+
+Configure the message encoding:
+
+* :kconfig:option:`CONFIG_LOG_BACKEND_NRF_CLOUD_OUTPUT_TEXT` or :kconfig:option:`CONFIG_LOG_BACKEND_NRF_CLOUD_OUTPUT_DICTIONARY`
+
+See `Dictionary-based Logging`_ to learn how dictionary-based logging works, how the dictionary is built, and how to decode the binary log output.
+
+Configure the default log level to be sent to the cloud:
+
+* :kconfig:option:`CONFIG_NRF_CLOUD_LOG_OUTPUT_LEVEL` set to 0 for NONE (to disable), 1 for ERR, 2 for WRN, 3 for INF, or 4 for DBG.
+
+For fine run-time control of log levels for each logging source, configure the following:
+
+* :kconfig:option:`CONFIG_LOG_RUNTIME_FILTERING`
+
+See `Run-time Filtering`_ for more information.
+
+Finally, configure these additional options:
+
+* :kconfig:option:`CONFIG_LOG_MODE_DEFERRED`
+* :kconfig:option:`CONFIG_LOG_PROCESS_THREAD_STACK_SIZE` set to 4096.
+* :kconfig:option:`CONFIG_LOG_BUFFER_SIZE` set to the maximum size of buffered log data before transmission to the cloud.
+* :kconfig:option:`CONFIG_LOG_PROCESS_THREAD_SLEEP_MS` set to the maximum time log messages can be buffered before transmission to the cloud.
+* :kconfig:option:`CONFIG_LOG_PRINTK` to n so that periodic messages from the lte_lc library do not get sent to the cloud.
```

Comment:
```suggestion
* :kconfig:option:`CONFIG_LOG_PRINTK` to ``n`` so that periodic messages from the :ref:`lte_lc_readme` library do not get sent to the cloud.
```